### PR TITLE
Review and submit a new circuit proposal

### PR DESCRIPTION
**Pull request by eloaverona**
This PR adds the review and submit part of the circuit proposal form and the logic to handle the submit request

### To test:
1. Run:
```
docker-compose -f docker/docker-compose.yaml up --build
```
2. To see the alpha node UI go to: `localhost:3030`. Log in or register a new user.

3. Click on the `circuits` option in the left side menu, it will forward you to  `localhost:3030/circuits`. The Propose New Circuit button will be disabled until the user has active keys set in their profile. This is a temporary solution until a more robust strategy can be developed.
Screen Shot 2020-06-21 at 11 15 44 AM
4. To set keys click in the "profile" icon in the bottom left corner of the screen to go to `localhost:3030/profile` and click the "+" button to add new keys:
Screen Shot 2020-06-21 at 11 21 39 AM
5. In the command line log into the generate-registry container to fetch alice's key:

```
docker-compose -f docker/docker-compose.yaml run generate-registry bash
cat /registry/alice.priv
cat /registry/alice.pub
```
6. Copy Alice's public and private key to the form and create the keys:
Screen Shot 2020-06-21 at 11 25 16 AM
7. Click on the key information to set it as active:
Screen Shot 2020-06-21 at 11 26 06 AM
8. Click on the `circuits` option in the left side menu, it will forward you to  `localhost:3030/circuits`. Now the Propose New Circuit button will be enabled:
Screen Shot 2020-06-21 at 11 27 04 AM
9. Click on the Propose New Circuit button to see the form. Select the Beta node:
Screen Shot 2020-06-21 at 11 28 43 AM

10. Click "Next" and add the service information:

1st service: 
   - service type = `scabbard`
   - service ID =`abcd`
   - allowed node = `alpha-node-000`
   - service arguments : 
             - key: `admin_keys`  value: `["<alice's public key>"]`
         - key: `peer_services` value: `["fghi"]`

2nd service:
   - service type = `scabbard`
   - service ID = `fghi`
   - allowed node = `beta-node-000`
   - service arguments : 
         - key: `admin_keys`  value: `["<alice's public key>"]`
         - key: `peer_services` value: `["abcd"]`

Screen Shot 2020-06-21 at 11 36 33 AM

11. Click "next". Add a circuit management type and a comment if you'd like: 
Screen Shot 2020-06-21 at 11 38 10 AM

12. Click "next". Add the application metadata if desired:
Screen Shot 2020-06-21 at 11 39 34 AM

13. Click "next". Review the information entered and if you're happy click submit! 

Screen Shot 2020-06-21 at 11 40 31 AM

14. If all goes well:
Screen Shot 2020-06-21 at 11 41 11 AM

15. You may need to refresh the page a couple times to see the proposal in the table. 
Screen Shot 2020-06-21 at 11 42 48 AM

16. If something goes wrong while submitting, the user will stay in the review page and have a chance to update the info they entered. In the example bellow I did not set up the scabbard service correctly:
Screen Shot 2020-06-21 at 11 44 30 AM

#### Extra:
Brownie tester points if you test that you can vote on your proposal successfully via the CLI.

1. Get Bob's private key

```
 docker-compose -f docker/docker-compose.yaml run generate-registry bash
 cat /registry/bob.priv
```

2. Connect to the beta splinter container:
```
docker-compose -f docker/docker-compose.yaml exec splinterd-beta bash
```

3. Save Bob's private key to a file:
```
echo "<bob's private key>" > bob.priv
```

4. List circuit proposals
```
splinter circuit proposals --url http://splinterd-beta:8085
```

5. Note the circuit ID for the proposal you created via the UI, submit a vote for it:
```
splinter circuit vote --key bob.priv --url http://splinterd-beta:8085 <CIRCUIT ID> --accept
```

6. Go to the back to the UI, `localhost:3030/circuits`. You may need to refresh the page a couple times. Notice that your circuit should not have the "awaiting approval" status label. 

#### Extra 2:
Extra brownie tester points if you test that you can submit a proposal via the UI from the Beta node (localhost:3031).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,6 +34,8 @@
 /sapling-dev-server/register-login.js
 /sapling-dev-server/profile
 
+/saplings/circuits/src/compiled_protos.json
```
Comment (agunde406) [inline .gitignore:37]: typo in commit message:  compite -> compile

**Review by agunde406 — CHANGES_REQUESTED** [review]
tested and was able to submit a circuit proposal. However the first time I did it, I messed up my keys and it failed silently on creating the payload. There should be a error handling there with an error toast so its clear what is going wrong.

File: saplings/circuits/src/components/ProposeCircuitButton.js [hunk 2]
```diff
@@ -26,8 +26,15 @@ const ProposeCircuitButton = () => {
     </span>
   );
 
+  const userHasKeys = JSON.parse(sessionStorage.getItem('CANOPY_KEYS'));
```
Comment (peterschwarz) [inline saplings/circuits/src/components/ProposeCircuitButton.js]: Isn't there a canopy function for this?

File: .gitignore (hunk 1 above)
Comment (eloaverona) [reply .gitignore:37]: Updated

File: saplings/circuits/src/components/ProposeCircuitButton.js (hunk 2 above)
Comment (eloaverona) [reply saplings/circuits/src/components/ProposeCircuitButton.js]: There is! Updated to use the function

**Comment by eloaverona** [comment]
> tested and was able to submit a circuit proposal. However the first time I did it, I messed up my keys and it failed silently on creating the payload. There should be a error handling there with an error toast so its clear what is going wrong.

@agunde406 Updated the PR to at least show an error message if it fails to build the payload. It might not super help the user to identify the actual issue, but at least they will see a response in the UI. Better error messages should be implemented in a later PR. I created a story for that.

I also updated to serialize the metadata depending on the encoding type. Before it was encoding everything as if the input was a utf8 string, which would not properly encode base64 metadata.

**Review by peterschwarz — APPROVED** [review]
Tested

File: saplings/circuits/src/components/forms/MultiStepForm.js [hunk 3]
```diff
@@ -50,8 +50,9 @@ export function MultiStepForm({
   };
 
   const submit = () => {
```
Comment (RyanLassigBanks) [inline saplings/circuits/src/components/forms/MultiStepForm.js:52]: Typo in commit message "sucess" -> success

**Review by agunde406 — APPROVED** [review]
Retested

**Pull request merged**